clang/bindings/python/tests/cindex/test_file.py

```diff
@@ -16,3 +17,16 @@ def test_file(self):
         self.assertEqual(str(file), "t.c")
         self.assertEqual(file.name, "t.c")
         self.assertEqual(repr(file), "<File: t.c>")
+
+    def test_file_eq(self):
+        path = os.path.join(kInputsDir, "hello.cpp")
+        header_path = os.path.join(kInputsDir, "header3.h")
+        tu = TranslationUnit.from_source(path)
```

Comment:
You should bring back the test which uses in-memory files, assert that files are not equal, and leave a FIXME that it shouldn't be the case. Would be nice if you read contents of those files from disk, instead of hard-coding them.

Reply:
I added the test with in-memory files first to check, because I suspect this might be a problem with those specifically... I'll abuse the CI for testing here since I can't reproduce the issue locally.
Will try using disk files for the test afterwards.

Reply:
> I'll abuse the CI for testing here since I can't reproduce the issue locally.

No worries